### ISSUE TITLE
Prod-v16.2.X

### ISSUE THREAD

**Issue by bortigno** [UFLX2MuMu/Ntupliser#115]
Tracking production/release v16.2.X.
Previous production/release work on #72 
Production linked to v18.2.X #114

**Comment by bortigno** [comment]
```
git tag -m "Test production v16.2.X" prod-v16.2.0
python crab/make_crab_script.py -s SingleMu_2016B H2Mu_gg ZJets_MG_1
# created the dir "/afs/cern.ch/user/b/bortigno/workspace/x2mm16_9413/src/Ntupliser/DiMuons/crab_2019_11_29_20_44-prod-v16.2.0/"
```
Now testing 
```
crab submit -c crab_2019_11_29_20_44-prod-v16.2.0/configs/H2Mu_gg.py --dryrun
```

First try there is an error. `CMSSW Return code: 8028`
It seems a problem with the input file so I tried to re-run the GRID credential..

It worked now.
Submitting the test prod.
```
crab proceed -d logs/crab_H2Mu_gg_2019_11_29_20_44_prod2016_prod-v16p2p0
./crab_2019_11_29_20_44-prod-v16.2.0/submit_all.sh 
```

**Comment by bortigno** [comment]
Checking the status of the jobs things seem to be going according to plans:

```
crab status -d logs/crab_ZJets_MG_1_2019_11_29_20_44_prod2016_prod-v16p2p0
CRAB project directory:		/afs/cern.ch/work/b/bortigno/x2mm16_9413/src/Ntupliser/DiMuons/logs/crab_ZJets_MG_1_2019_11_29_20_44_prod2016_prod-v16p2p0
Task name:			191129_215813:bortigno_crab_ZJets_MG_1_2019_11_29_20_44_prod2016_prod-v16p2p0
Grid scheduler - Task Worker:	crab3@vocms0195.cern.ch - crab-prod-tw01
Status on the CRAB server:	SUBMITTED
Task URL to use for HELP:	https://cmsweb.cern.ch/crabserver/ui/task/191129_215813%3Abortigno_crab_ZJets_MG_1_2019_11_29_20_44_prod2016_prod-v16p2p0
Dashboard monitoring URL:	https://monit-grafana.cern.ch/d/cmsTMDetail/cms-task-monitoring-task-view?orgId=11&var-user=bortigno&var-task=191129_215813%3Abortigno_crab_ZJets_MG_1_2019_11_29_20_44_prod2016_prod-v16p2p0
In case of issues with the dashboard, please provide feedback to hn-cms-computing-tools@cern.ch
Status on the scheduler:	SUBMITTED

Jobs status:                    idle         		 25.5% ( 36/141)
				unsubmitted  		 74.5% (105/141)

No publication information (publication has been disabled in the CRAB configuration file)
Log file is /afs/cern.ch/work/b/bortigno/x2mm16_9413/src/Ntupliser/DiMuons/logs/crab_ZJets_MG_1_2019_11_29_20_44_prod2016_prod-v16p2p0/crab.log
```

**Comment by bortigno** [comment]
Production went medium

```
for d in logs/crab_*2019_11*; do crab status -d ${d}; done

CRAB project directory:		/afs/cern.ch/work/b/bortigno/x2mm16_9413/src/Ntupliser/DiMuons/logs/crab_H2Mu_gg_2019_11_29_20_44_prod2016_prod-v16p2p0
Task name:			191129_195213:bortigno_crab_H2Mu_gg_2019_11_29_20_44_prod2016_prod-v16p2p0
Grid scheduler - Task Worker:	crab3@vocms0106.cern.ch - crab-prod-tw01
Status on the CRAB server:	SUBMITTED
Task URL to use for HELP:	https://cmsweb.cern.ch/crabserver/ui/task/191129_195213%3Abortigno_crab_H2Mu_gg_2019_11_29_20_44_prod2016_prod-v16p2p0
Dashboard monitoring URL:	https://monit-grafana.cern.ch/d/cmsTMDetail/cms-task-monitoring-task-view?orgId=11&var-user=bortigno&var-task=191129_195213%3Abortigno_crab_H2Mu_gg_2019_11_29_20_44_prod2016_prod-v16p2p0
In case of issues with the dashboard, please provide feedback to hn-cms-computing-tools@cern.ch
Status on the scheduler:	COMPLETED

Jobs status:                    finished     		100.0% (6/6)

No publication information (publication has been disabled in the CRAB configuration file)

Warning: the max jobs runtime is less than 30% of the task requested value (1250 min), please consider to request a lower value for failed jobs (allowed through crab resubmit) and/or improve the jobs splitting (e.g. config.Data.splitting = 'Automatic') in a new task.

Summary of run jobs:
 * Memory: 1359MB min, 1394MB max, 1372MB ave
 * Runtime: 0:22:52 min, 2:03:21 max, 1:06:00 ave
 * CPU eff: 86% min, 96% max, 93% ave
 * Waste: 0:00:57 (0% of total)

Log file is /afs/cern.ch/work/b/bortigno/x2mm16_9413/src/Ntupliser/DiMuons/logs/crab_H2Mu_gg_2019_11_29_20_44_prod2016_prod-v16p2p0/crab.log
CRAB project directory:		/afs/cern.ch/work/b/bortigno/x2mm16_9413/src/Ntupliser/DiMuons/logs/crab_SingleMu_2016B_2019_11_29_20_44_prod2016_prod-v16p2p0
Task name:			191129_215749:bortigno_crab_SingleMu_2016B_2019_11_29_20_44_prod2016_prod-v16p2p0
Grid scheduler - Task Worker:	crab3@vocms0106.cern.ch - crab-prod-tw01
Status on the CRAB server:	SUBMITTED
Task URL to use for HELP:	https://cmsweb.cern.ch/crabserver/ui/task/191129_215749%3Abortigno_crab_SingleMu_2016B_2019_11_29_20_44_prod2016_prod-v16p2p0
Dashboard monitoring URL:	https://monit-grafana.cern.ch/d/cmsTMDetail/cms-task-monitoring-task-view?orgId=11&var-user=bortigno&var-task=191129_215749%3Abortigno_crab_SingleMu_2016B_2019_11_29_20_44_prod2016_prod-v16p2p0
In case of issues with the dashboard, please provide feedback to hn-cms-computing-tools@cern.ch
Status on the scheduler:	FAILED

Jobs status:                    failed       		 45.2% (238/527)
				finished     		 54.8% (289/527)

No publication information (publication has been disabled in the CRAB configuration file)

Error Summary: (use crab status --verboseErrors for details about the errors)

174 jobs failed in postprocessing step

 18 jobs failed with exit code 8028

 16 jobs failed with exit code 50660

  9 jobs failed with exit code 50664

  4 jobs failed with exit code 8021

Could not find exit code details for 17 jobs.

Have a look at https://twiki.cern.ch/twiki/bin/viewauth/CMSPublic/JobExitCodes for a description of the exit codes.

Summary of run jobs:
 * Memory: 4MB min, 2623MB max, 1290MB ave
 * Runtime: 0:01:02 min, 15:13:08 max, 2:59:26 ave
 * CPU eff: 0% min, 99% max, 93% ave
 * Waste: 1489:23:54 (49% of total)

Log file is /afs/cern.ch/work/b/bortigno/x2mm16_9413/src/Ntupliser/DiMuons/logs/crab_SingleMu_2016B_2019_11_29_20_44_prod2016_prod-v16p2p0/crab.log
CRAB project directory:		/afs/cern.ch/work/b/bortigno/x2mm16_9413/src/Ntupliser/DiMuons/logs/crab_ZJets_MG_1_2019_11_29_20_44_prod2016_prod-v16p2p0
Task name:			191129_215813:bortigno_crab_ZJets_MG_1_2019_11_29_20_44_prod2016_prod-v16p2p0
Grid scheduler - Task Worker:	crab3@vocms0195.cern.ch - crab-prod-tw01
Status on the CRAB server:	SUBMITTED
Task URL to use for HELP:	https://cmsweb.cern.ch/crabserver/ui/task/191129_215813%3Abortigno_crab_ZJets_MG_1_2019_11_29_20_44_prod2016_prod-v16p2p0
Dashboard monitoring URL:	https://monit-grafana.cern.ch/d/cmsTMDetail/cms-task-monitoring-task-view?orgId=11&var-user=bortigno&var-task=191129_215813%3Abortigno_crab_ZJets_MG_1_2019_11_29_20_44_prod2016_prod-v16p2p0
In case of issues with the dashboard, please provide feedback to hn-cms-computing-tools@cern.ch
Status on the scheduler:	FAILED

Jobs status:                    failed       		100.0% (141/141)

No publication information (publication has been disabled in the CRAB configuration file)

Error Summary: (use crab status --verboseErrors for details about the errors)

110 jobs failed in postprocessing step

 23 jobs failed with exit code 50664

  3 jobs failed with exit code 8028

  1 jobs failed with exit code 8021

Could not find exit code details for 4 jobs.

Have a look at https://twiki.cern.ch/twiki/bin/viewauth/CMSPublic/JobExitCodes for a description of the exit codes.

Summary of run jobs:
 * Memory: 44MB min, 2384MB max, 1303MB ave
 * Runtime: 0:01:37 min, 20:52:19 max, 12:00:16 ave
 * CPU eff: 0% min, 102% max, 93% ave
 * Waste: 2573:55:21 (60% of total)

Log file is /afs/cern.ch/work/b/bortigno/x2mm16_9413/src/Ntupliser/DiMuons/logs/crab_ZJets_MG_1_2019_11_29_20_44_prod2016_prod-v16p2p0/crab.log
```

The problem is very likely the lack of space.

**Comment by bortigno** [comment]
I cleared up some space in my EOS dir ( see #114 ).

I also back port from 2018 some useful update to the scripts and configuration

```
git cherry-pick 0750fe7
[prod-v16.2.X ce34736] lower the number of files per jobs after 'too much wall-time' error occurred in test run
 Date: Tue Dec 3 14:52:25 2019 +0100
 1 file changed, 1 insertion(+), 1 deletion(-)

git cherry-pick 445c468
[prod-v16.2.X 6e408ff] BUGFIX: fix check-all part of the script
 Date: Tue Dec 3 15:08:10 2019 +0100
 1 file changed, 1 insertion(+), 1 deletion(-)
```

**Comment by bortigno** [comment]
Trying to relaunch a test

```
python crab/make_crab_script.py -s SingleMu_2016B H2Mu_gg ZJets_MG_1
./crab_2019_12_03_18_09-prod-v16.2.0-2-g6e408ff/submit_all.sh 
./crab_2019_12_03_18_09-prod-v16.2.0-2-g6e408ff/check_all.sh 
```

Jobs are running.

**Comment by bortigno** [comment]
Removing the 2018 production that is replicated in the group space
```
rm -rf /eos/cms/store/user/bortigno/h2mm/ntuples/2018/102X/prod-v18.1.6.skim3l*
```

**Comment by bortigno** [comment]
`prod-v16.2.0-2-g6e408ff` didn't go very well. Most jobs failed in the post processing step. Successful jobs were 8.5% in ZJets, 92.9% for ggH, and 34.0% for signal with more than half jobs failed in the post processing step. The second largest cause of failed jobs are 8001, 50664, 8021 for both data and MC.

>8001 - Other CMS Exception
50664 - Application terminated by wrapper because using too much Wall Clock time
ALERT! 8021 - FileReadError (May be a site error)
The following link can help, please check: https://twiki.cern.ch/twiki/bin/view/CMSPublic/WorkBookWhichRelease#DifferentReleases

The 50664 should be fixed once moved to "Automatic" splitting, and the 8021 is a site error.
The 8001 is a bit more concerning. I can investigate further later.

**Comment by bortigno** [comment]
Backporting some extra options from 2018 for crab/make_crab_script.py

```
git cherry-pick 40c3a84
>[prod-v16.2.X 9c9791b] Trying automated job splitting
 Date: Thu Dec 5 16:18:49 2019 +0100
 1 file changed, 6 insertions(+), 6 deletions(-)
```
git cherry-pick fa70119880148aa137cb318d92b88440db6a3c20
>[prod-v16.2.X fba7b46] Setting longer jobs
 Date: Fri Dec 6 11:23:37 2019 +0100
 1 file changed, 2 insertions(+), 2 deletions(-)
```
git cherry-pick 3b89b5f
```
after fixing some minor conflicts
>[prod-v16.2.X 66c1f7d] Adding testing options and username fetching for output dir.
 Date: Fri Dec 6 12:32:03 2019 +0100
 1 file changed, 18 insertions(+), 14 deletions(-)

**Comment by bortigno** [comment]
@eyigitba will produce 2016

**Comment by eyigitba** [comment]
Started production with tag `prod-v16.2.1.skim3l`. 

There are already some failed jobs:

```
Task name:			191209_162131:eyigitba_crab_ZJets_AMC_2019_12_09_16_59_prod2016_prod-v16p2p1pskim3l
Failure message from server:	Block /DYJetsToLL_M-50_TuneCUETP8M1_13TeV-amcatnloFXFX-pythia8/RunIISummer16MiniAODv3-PUMoriond17_94X_mcRun2_asymptotic_v3_ext2-v1/MINIAODSIM#be00ed73-902f-4922-9387-74825818106f contains more than 100000 lumis.

Task name:			191209_162206:eyigitba_crab_ZJets_MG_1_2019_12_09_16_59_prod2016_prod-v16p2p1pskim3l
Failure message from server:	Block /DYJetsToLL_M-50_TuneCUETP8M1_13TeV-madgraphMLM-pythia8/RunIISummer16MiniAODv3-PUMoriond17_94X_mcRun2_asymptotic_v3_ext2-v2/MINIAODSIM#01598cae-c99b-4c92-9968-f3186fd2444e contains more than 100000 lumis.

Task name:			191209_162239:eyigitba_crab_ZJets_MG_2_2019_12_09_16_59_prod2016_prod-v16p2p1pskim3l
Failure message from server:	Block /DYJetsToLL_M-50_TuneCUETP8M1_13TeV-madgraphMLM-pythia8/RunIISummer16MiniAODv3-PUMoriond17_94X_mcRun2_asymptotic_v3_ext1-v2/MINIAODSIM#3d0ccb47-7e05-409d-a994-222a7062f983 contains more than 100000 lumis.

Task name:			191209_162942:eyigitba_crab_WZ_2l_2019_12_09_16_59_prod2016_prod-v16p2p1pskim3l
Failure message from server:	Block /WZTo2L2Q_13TeV_amcatnloFXFX_madspin_pythia8/RunIISummer16MiniAODv3-PUMoriond17_94X_mcRun2_asymptotic_v3-v2/MINIAODSIM#80512f19-d8af-4a20-adb8-99416bde6e36 contains more than 100000 lumis.
```

How to fix is listed here: https://twiki.cern.ch/twiki/bin/view/CMSPublic/CRAB3FAQ#crab_submit_fails_with_Block_AN1

Another job failed with exit code `8021` which means `FileReadError`:

```
1 jobs failed with following error message: (for example, job 1-1)
 		CmsRunFailure
 		CMSSW error message follows.
 		Fatal Exception
 		An exception of category 'FileReadError' occurred while
 		  [0] Rethrowing an exception that happened on a different thread.
 		  [1] Reading branch EcalRecHitsSorted_reducedEgamma_reducedEBRecHits_PAT.
 		  [2] Calling XrdFile::readv()
 		  [3] XrdAdaptor::ClientRequest::HandleResponse() failure while running connection recovery
 		  [4] Handling XrdAdaptor::RequestManager::requestFailure()
 		  [5] In XrdAdaptor::RequestManager::OpenHandler::HandleResponseWithHosts()
 		Exception Message:
 		XrdCl::File::Open(name='root://xrootd.t2.ucsd.edu:2040//store/mc/RunIISummer16MiniAODv3/TTWW_TuneCUETP8M2T4_13TeV-madgraph-pythia8/MINIAODSIM/PUMoriond17_94X_mcRun2_asymptotic_v3_ext1-v1/30000/FC756CE3-8E32-E911-B404-FA163EEEB0C7.root', flags=0x10, permissions=0660) => error '[ERROR] Server responded with an error: [3011] Cluster retry limit exceeded.
 		' (errno=3011, code=400)
 		  Additional Info:
 		      [a] Original error: '[ERROR] Operation expired' (errno=0, code=206, source=xcache-11.t2.ucsd.edu:1094 (site UCSD-XCACHE)).
 		      [b] Original failed source is xcache-11.t2.ucsd.edu:1094 (site UCSD-XCACHE)
 		      [c] Disabled source: xcache-11.t2.ucsd.edu:1094
```

I'll probably resubmit these jobs once the rest is completed.

**Comment by eyigitba** [comment]
First batch of production is mostly completed. There is a good amount of failed jobs. They will be resubmitted once everything is finished. Below is a list of the fail codes and task names.

```
SingleMu_2016B:
178 jobs failed with exit code 8002
Could not find exit code details for 7 jobs.

SingleMu_2016C:
1 jobs failed with exit code 50660
 1 jobs failed with exit code 8002

SingleMu_2016D:
 96 jobs failed with exit code 8002

SingleMu_2016E:
3 jobs failed with exit code 8002
  1 jobs failed with exit code 50660

SingleMu_2016F:
 2 jobs failed with exit code 50660

SingleMu_2016G:
161 jobs failed with exit code 8002
Could not find exit code details for 1 jobs.

SingleMu_2016H:
155 jobs failed with exit code 8002
Could not find exit code details for 18 jobs.

ZZ_4l:
 16 jobs failed with exit code 8028

ttWW:
1 jobs failed with exit code 8021

Zjets_AMC:
Failure message from server:	Block /DYJetsToLL_M-50_TuneCUETP8M1_13TeV-amcatnloFXFX-pythia8/RunIISummer16MiniAODv3-PUMoriond17_94X_mcRun2_asymptotic_v3_ext2-v1/MINIAODSIM#be00ed73-902f-4922-9387-74825818106f contains more than 100000 lumis.

Zjets_MG1:
Failure message from server:	Block /DYJetsToLL_M-50_TuneCUETP8M1_13TeV-madgraphMLM-pythia8/RunIISummer16MiniAODv3-PUMoriond17_94X_mcRun2_asymptotic_v3_ext2-v2/MINIAODSIM#01598cae-c99b-4c92-9968-f3186fd2444e contains more than 100000 lumis.

Zjets_MG2:
Failure message from server:	Block /DYJetsToLL_M-50_TuneCUETP8M1_13TeV-madgraphMLM-pythia8/RunIISummer16MiniAODv3-PUMoriond17_94X_mcRun2_asymptotic_v3_ext1-v2/MINIAODSIM#3d0ccb47-7e05-409d-a994-222a7062f983 contains more than 100000 lumis.

WZ_2l:
Failure message from server:	Block /WZTo2L2Q_13TeV_amcatnloFXFX_madspin_pythia8/RunIISummer16MiniAODv3-PUMoriond17_94X_mcRun2_asymptotic_v3-v2/MINIAODSIM#80512f19-d8af-4a20-adb8-99416bde6e36 contains more than 100000 lumis.
```

`ZJets_hiM_AMC` is still running.

**Comment by eyigitba** [comment]
Production of 2016 3 lepton skim ntuples is completed. There were failing jobs on 2 datasets that couldn't be resubmitted. The tasks for following datasets were killed and resubmitted with a new lumi mask to only produce the missing lumi sections: 

```
SingleMu_2016B
SingleMu_2016D
```
This is done as described here:

```
https://twiki.cern.ch/twiki/bin/view/CMSPublic/CRAB3FAQ#Dealing_with_a_growing_input_dat
```
The only difference is we are just using the missinLumi.py file instead of taking differences between lumiMasks.

After the production of the new tasks is completed, I 'hadd'ed all the files 5 files to 1 and moved them to higgs group space without removing the non-hadded files. We can remove them in the future if need be. While doing this I copied the `SingleMu_2016B` and `SingleMu_2016D` in the most recent date directory and hadded them there. The location of the files are below:

```
/eos/cms/store/group/group_higgs/HiggsExo/H2Mu/UF/ntuples/2016/94X_v4/prod-v16.2.1.skim3l/
```
